cpp/tests/test_ide_parameters_io.cpp

```diff
@@ -52,88 +62,120 @@ TEST(TestIDEParametersIo, RKIcompareWithPreviousRun)
-    EXPECT_NEAR(model.m_total_confirmed_cases, 70, 1e-4);
+    std::vector<ScalarType> deaths = {
+        1, 2.471428571455, 26.34999999999, 603.621428571465, 3972.41428571431, 7668.84999999998};
 
-    ASSERT_EQ(compare.size(), static_cast<size_t>(model.m_transitions.get_num_time_points()));
-    for (size_t i = 0; i < compare.size(); i++) {
-        ASSERT_EQ(compare[i].size(), static_cast<size_t>(model.m_transitions.get_num_elements()) + 1) << "at row " << i;
-        ASSERT_NEAR(compare[i][0], model.m_transitions.get_time(i), 1e-7) << "at row " << i;
-        for (size_t j = 1; j < compare[i].size(); j++) {
-            ASSERT_NEAR(compare[i][j], model.m_transitions.get_value(i)[j - 1], 1e-7) << " at row " << i;
-        }
+    std::vector<ScalarType> total_confirmed_cases = {10269.2857142857, 29615.8571428571, 185321.571428571,
+                                                     215386.428571429, 77163.5714285714, 35588.4285714286};
+
+    for (mio::AgeGroup group = mio::AgeGroup(0); group < mio::AgeGroup(num_agegroups); ++group) {
+        int Di = model.get_state_flat_index((Eigen::Index)mio::isecir::InfectionState::Dead, group);
+        EXPECT_NEAR(model.m_populations.get_value(0)[Di], deaths[size_t(group)], 1e-4);
+        EXPECT_NEAR(model.m_total_confirmed_cases[group], total_confirmed_cases[size_t(group)], 1e-4);
+    }
+
+    // Compare transitions at last time point with results from a previous run that are given here.
+    mio::Vector<ScalarType> compare(num_transitions * num_agegroups);
+    compare << 336.428571428600, 328.285714285701, 162.000000000000, 163.071428571425, 80.130989648839, 79.803571428575,
+        39.476374533415, 39.476374533415, 19.550404043081, 19.550404043081, 1105.714285714297, 1069.857142857200,
+        515.714285714250, 163.071428571425, 80.130989648839, 79.803571428575, 39.476374533415, 39.476374533415,
+        19.550404043081, 19.550404043081, 5819.000000000000, 5744.000000000000, 2806.428571428551, 163.071428571425,
+        80.130989648839, 79.803571428575, 39.476374533415, 39.476374533415, 19.550404043081, 19.550404043081,
+        6685.142857142899, 6572.857142857101, 3200.714285714304, 163.071428571425, 80.130989648839, 79.803571428575,
+        39.476374533415, 39.476374533415, 19.550404043081, 19.550404043081, 2376.000000000000, 2342.285714285696,
+        1142.571428571406, 163.071428571425, 80.130989648839, 79.803571428575, 39.476374533415, 39.476374533415,
+        19.550404043081, 19.550404043081, 966.714285714304, 946.142857142797, 457.214285714301, 163.071428571425,
+        80.130989648839, 79.803571428575, 39.476374533415, 39.476374533415, 19.550404043081, 19.550404043081;
+
+    mio::isecir::Simulation sim(model, dt);
+    int last_time_point = int(model.m_transitions.get_last_time());
+    ASSERT_EQ(compare.size(), model.m_transitions[last_time_point].size());
+
+    for (int j = 1; j < compare.size(); j++) {
```

Comment:
```suggestion
    for (int j = 0; j < compare.size(); j++) {
```
When defining `compare` like above, we don't have the time stored anywhere and the relevant entries start at index 0. Same in test below.